lib/ruby_lsp/extensions/base.rb

```diff
@@ -0,0 +1,55 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Extensions
+    class Base
```

Comment:
I don't think I like the `RubyLsp::Extensions::Base` name.

A better name structure would be `RubyLsp::Extension` class that extension writers can subclass to define their `activate` method, and then `RubyLsp::Provider::Hover` as the class to inherit for a hover provider. Since the hover provider "provides" hover content, but is not the "extension" itself.